frontend/src/core/placeable/parsers/numberString.js

```diff
@@ -18,7 +18,7 @@ import { Localized } from '@fluent/react';
  * https://github.com/translate/translate/blob/2.3.1/translate/storage/placeables/general.py#L72
  */
 const numberString = {
-    rule: /([-+]?[0-9]+([\u00A0.,][0-9]+)*)/u,
+    rule: /([-+]?[0-9]+([\u00A0.,][0-9]+)*)\b/u,
```

Comment:
That's a valid fix. But "3D", "A4" and the likes are candidates for placeables, so I wonder if we should make `allCapitalsString` catch them?

Reply:
BTW, how hard would it be to make `allCapitalsString` not catch this at all?
https://pontoon.mozilla.org/translate/sl/firefox/all-resources/?string=204038

I mean, would it be easier than catching all Latin characters? Maybe we can require whitespace/start/end around the capitals.

Reply:
That would mean we wouldn't catch `CAPS.` for example, because it's followed by a dot, which is effectively a word boundary but not an empty space. We would need to add something very close to the `\b` word boundary rule but not exactly, and I'm not sure that's wise to do as it's bound to lead to other bugs. And to a very complex regepx anyway.

I don't think there's a good way out of this problem with the current state of RegExp in JavaScript. We just need the browsers to improve their support of unicode. :/